bussiness-logic-module/src/main/java/edu/hillel/service/UserService.java

```diff
@@ -29,16 +33,34 @@ public static UserService getSingletonInstance(UserRepository userRepository) {
     }
 
     public User logIn(String login, String password) {
-        loggedInUser = userRepository
+        User validUser = userRepository
                 .findUserByLoginPassword(login, password)
                 .orElseThrow(() -> new IllegalArgumentException("There no user with such login and password"));
-        return loggedInUser;
+        loggedInUser = validUser; // for old realisation with one user
+        return validUser;
+    }
+
+    public void checkAlreadyLoggedUser(Map<String, HttpSession> loggedInUsers, String userLogin) throws AccessException {
+        HttpSession httpSession = loggedInUsers.get(userLogin);
+        if (httpSession != null) {
```

Comment:
wouldn`t it be easier to implement redirection by returning enum value not throwing exception. Your approach works but it looks dirty and returnig enum with following switch-case processing would look cleaner

Reply:
As I correct understand you wrote about two cases catching exception in class LoginServlet in method doPost.
In the method  checkAlreadyLoggedUser . In servlet no direct method to check if current session is valid, so I use httpSession.getAttribute("user"); and if no such Attribute in session, method throw one exception IllegalStateException. And I skip this exception because user is logout. but if no I throw another exception. this method i made void, so I can return some type. But method logIn(String login, String password)  that used in Servlet doPost also , returned type User, so I cannot return enum value. Am I correct understand you?